### PR TITLE
Update documentation and mark HTTP client class as isolated

### PR DESCRIPTION
**Pull request by LakshanSS**
## Purpose
- Fix https://github.com/wso2-enterprise/choreo/issues/6677
- Update documentation to new format

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
JDK 11
SL Beta2

### CODE REVIEW

**Comment by Chuhaa** [comment]
@LakshanSS  Can you add comments for listener `init`?

**Comment by LakshanSS** [comment]
> @LakshanSS Can you add comments for listener `init`?

Done

File: README.md [hunk 1]
```diff
@@ -1,236 +1,22 @@
...
-    }
-}
-```
-## Please check the [Samples directory](https://github.com/ballerina-platform/module-ballerinax-slack/tree/master/samples) for more examples.
-
-
-## Module Overview - `ballerinax/slack.'listener`
-
-The `ballerinax/slack.'listener` module provides a Listener to grasp event triggers from your Slack App. This functionality is provided by [Slack Event API](https://api.slack.com/apis/connections/events-api). 
+This connector allows you to access the Slack Web API and Slack Events API through Ballerina.
```
Comment (shaniR) [inline README.md:10]: It provides the capability to query information and perform some actions in a Slack workspace. It also allows you to listen to Slack Events. -> some actions? shall we specify what they are?

File: README.md [hunk 2]
```diff
@@ -240,40 +26,36 @@ service /slack on slackListener {
...
+2. Download and install [Ballerina Swan Lake Beta2](https://ballerina.io/). 
 
-### Building the Source
+### Building the source
 
-Execute the commands below to build from the source after installing Ballerina SL Alpha 5.
+Execute the commands below to build from the source.
 
-1. To build the ballerina package:
+1. To build the gradle project:
```
Comment (shaniR) [inline README.md:35]: Gradle

File: README.md [hunk 3]
```diff
@@ -240,40 +26,36 @@ service /slack on slackListener {
...
 ## Contributing to Ballerina
-
 As an open source project, Ballerina welcomes contributions from the community. 
 
 For more information, go to the [contribution guidelines](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md).
 
-## Code of Conduct
-
+## Code of conduct
 All the contributors are encouraged to read the [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
```
Comment (shaniR) [inline README.md:56]: Use the sentence in the guideline.

File: slack/Module.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+## Overview
+This module allows you to access the Slack Web API through Ballerina. This module provides the capability
```
Comment (shaniR) [inline slack/Module.md:2]: to query information and

File: slack/Module.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+to query information from and perform some actions in a Slack workspace.
+
+## Prerequisites
+Before using this connector in your Ballerina application, complete the following:
+
+* Create a Slack account.
+    1. Visit https://slack.com/get-started#/createnew and create a Slack account.
+* Obtain Slack User Oauth Token.
+    1. Visit https://api.slack.com/apps and create a Slack App.
```
Comment (shaniR) [inline slack/Module.md:11]: step 1 is repeated. Fix numbering.

Comment (shaniR) [reply slack/Module.md:11]: Is there any document for instruction on how to obtain Slack User OAuth token?

Comment (shaniR) [reply slack/Module.md:11]: OAuth 2.0

File: slack/Module.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    2. In the `Add features and functionality` section, Click `Permissions`.
+    3. Go to the `Scopes` section and add necessary OAuth scopes in `User Token Scopes` section. ("channels:history", "channels:read", "channels:write", "chat:write", "emoji:read", files:read", "files:write", "groups:read", "reactions:read", "users:read", "users:read.email")
```
Comment (shaniR) [inline slack/Module.md:13]: OAuth 2.0

File: slack/Module.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    4. Go back to `Basic Information` section of your Slack App, go to `Install your app` section and Install the app to the workspace by clicking `Install to Workspace` button.
```
Comment (shaniR) [inline slack/Module.md:14]: Go back to `Basic Information` section of your Slack App. Then go to `Install your app` section and install the app to the workspace by clicking `Install to Workspace` button.

File: slack/Module.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    5. Get your User OAuth token from the `OAuth & Permissions` section of your Slack App.
```
Comment (shaniR) [inline slack/Module.md:15]: OAuth 2.0

File: slack/Module.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+## Quickstart
+
+To use the Slack connector in your Ballerina application, update the .bal file as follows:
+
+### Step 1: Import connector
+First, import the `ballerinax/slack` module as shown below.
```
Comment (shaniR) [inline slack/Module.md:22]: let's remove the word First,

File: slack/Package.md [hunk 10]
```diff
@@ -1,221 +1,21 @@
...
-# Ballerina Slack Connector 
-Connects to Slack using Ballerina
+## Package Overview
+This is a [Ballerina](https://ballerina.io/) connector for Slack.
 
-# Introduction
-## Slack
-[Slack](https://api.slack.com/) is a channel-based messaging platform. With Slack, people can work together more effectively, connect all their software tools and services, and find the information they need to do their best work — all within a secure, enterprise-grade environment.
+This package allows you to access the Slack Web API and Slack Events API through Ballerina.
+It provides the capability to query information from and perform some actions in a Slack workspace. It also allows you to listen to Slack Events.
```
Comment (shaniR) [inline slack/Package.md:7]: query information and
again, we need to mention what the some operations are.

**Review by shaniR — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (LakshanSS) [reply README.md:35]: done

File: README.md (hunk 3 above)
Comment (LakshanSS) [reply README.md:56]: done

File: slack/Module.md (hunk 4 above)
Comment (LakshanSS) [reply slack/Module.md:2]: done

File: slack/Module.md (hunk 5 above)
Comment (LakshanSS) [reply slack/Module.md:11]: > step 1 is repeated. Fix numbering.

Nope. Creating a Slack **Account** and Slack **App** are 2 different things. :)

Comment (LakshanSS) [reply slack/Module.md:11]: > Is there any document for instruction on how to obtain Slack User OAuth token?

No. There is no documentation provided by Slack. Thats why we have to give the steps here.

Comment (LakshanSS) [reply slack/Module.md:11]: > OAuth 2.0

This OAuth is not the `OAuth 2.0`. This is a name give to their token by Slack.
Please refer the following screenshot which shows the name of the token.

sss

File: slack/Module.md (hunk 6 above)
Comment (LakshanSS) [reply slack/Module.md:13]: We are referring to the name used by Slack. Screenshot attached below. 

scopeOauth

File: slack/Module.md (hunk 8 above)
Comment (LakshanSS) [reply slack/Module.md:15]: We are using the term used by Slack. Screenshot attached.

oauthperm

File: slack/Package.md (hunk 10 above)
Comment (LakshanSS) [reply slack/Package.md:7]: Fixed the first issue.

Second one -> The number of operations we support is a big list. Those details are available in the api docs. We don't need to repeat it here. In Choreo, those operations will be shown in a dropdown list. Users can refer those places.

If we add them here, in the future when we add new operations, we'll have to edit here also. So better not to mention them here.

File: slack/Module.md (hunk 9 above)
Comment (LakshanSS) [reply slack/Module.md:22]: done

File: slack/Module.md (hunk 7 above)
Comment (LakshanSS) [reply slack/Module.md:14]: done

File: README.md (hunk 1 above)
Comment (LakshanSS) [reply README.md:10]: That is a big list. That list is available in the api docs. In choreo, those will be listed in a dropdown.
I think we don't need to repeat them here.

File: slack/Module.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
+## Overview
+This module allows you to access the Slack Web API through Ballerina. This module provides the capability
+to query information and perform some actions in a Slack workspace.
+
```
Comment (shaniR) [inline slack/Module.md:4]: The module version is not mentioned.

Comment (LakshanSS) [reply slack/Module.md:4]: Yeah. For all the connectors, we specify the version of the API.
But for Slack, they haven't any version for their apis.
Thats why we couldn't specify the API version

File: README.md (hunk 1 above)
Comment (abeykoon) [reply README.md:10]: Shall we specify a few and say etc.

File: slack/Module.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    3. Go to the `Scopes` section and add necessary OAuth scopes in `User Token Scopes` section. ("channels:history", "channels:read", "channels:write", "chat:write", "emoji:read", files:read", "files:write", "groups:read", "reactions:read", "users:read", "users:read.email")
+    4. Go back to `Basic Information` section of your Slack App. Then go to `Install your app` section and install the app to the workspace by clicking `Install to Workspace` button.
+    5. Get your User OAuth token from the `OAuth & Permissions` section of your Slack App.
+
+## Quickstart
+
+To use the Slack connector in your Ballerina application, update the .bal file as follows:
+
+### Step 1: Import connector
+Import the `ballerinax/slack` module as shown below.
```
Comment (abeykoon) [inline slack/Module.md:22]: Please double check the module we need to use for listener.

**Review by abeykoon — APPROVED** [review]

File: slack/endpoint.bal [hunk 13]
```diff
@@ -16,12 +16,18 @@
 
 import ballerina/http;
 
-# Client for Slack connector.  
+# Ballerina Slack connector provides the capabiliy to access Slack Web API.
+# This connector lets you access the Slack Web API using a Slack User Oath token.
```
Comment (shaniR) [inline slack/endpoint.bal:20]: OAuth 2.0?

File: slack/endpoint.bal [hunk 14]
```diff
@@ -16,12 +16,18 @@
...
 @display {label: "Slack", iconPath: "logo.png"}
-public client class Client {
+public isolated client class Client {
     private map<string> channelIdMap = {};
-    private http:Client slackClient;
+    private final http:Client slackClient;
 
+    # Initializes the connector. During initialization, you have to pass a Slack User Oauth Token.
```
Comment (shaniR) [inline slack/endpoint.bal:26]: To initialize,

Comment (shaniR) [reply slack/endpoint.bal:26]: OAuth 2.0

File: slack/endpoint.bal [hunk 15]
```diff
@@ -16,12 +16,18 @@
...
+    # Visit https://api.slack.com/apps and create a Slack App.
+    # Obtain your User OAuth token from the `OAuth & Permissions` section of your Slack App.
```
Comment (shaniR) [inline slack/endpoint.bal:28]: OAuth 2.0

**Pull request merged**